pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -229,7 +229,7 @@ func (d *defaultDriver) newElasticsearchClient(
 	v version.Version,
 	caCerts []*x509.Certificate,
 ) esclient.Client {
-	url := fmt.Sprintf("https://%s.%s.svc:%d", service.Name, service.Namespace, network.HTTPPort)
+	url := fmt.Sprintf("%s://%s.%s.svc:%d", d.ES.Spec.HTTP.Scheme(), service.Name, service.Namespace, network.HTTPPort)
```

Comment:
I think we need to be smarter if we want to handle TLS <-> non-TLS migrations. Rolling upgrade would probably fail since we cannot request the cluster.

Reply:
Indeed!

This PR is still relatively small, we could try to address it right here, wdyt?

Reply:
up to you :)